scripts/zones/Carpenters_Landing/mobs/Para.lua

```diff
@@ -0,0 +1,97 @@
+-----------------------------------
+--  Area: Carpenters' Landing
+--    NM: Para
+-- Quest: Elderly Pursuits
+-----------------------------------
+local ID = zones[xi.zone.CARPENTERS_LANDING]
+-----------------------------------
+---@type TMobEntity
+local entity = {}
+
+entity.onMobSpawn = function(mob)
+    mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180) -- 3 minutes
+
+    local para = GetMobByID(ID.mob.PARA)
+
+    if
+        Para:getLocalVar('spawnAdds') < 5
+    then
+        Para:setLocalVar('spawnAdds', Para:getLocalVar('spawnAdds') + 1)
+    end
+
+end
+
+entity.onMobFight = function(mob, target)
+end
+
+--If Para uses TP specific TP moves move under 50% HP then it will spawn a clone.
+--Will only spawn 4 adds total.
+--Adds will spawn ontop of the player.
+--TP moves that spawned a clone in the capture: 310 - Queezyshroom, 311- Numbshroom, 312 - Shakeshroom
+--No reports or captures if the clones can respawn.
+--Reports from players saying only the original needs to be killed.
+entity.onMobWeaponSkill = function(target, mob, skill, player)
+    local hpp   = mob:getHPP()
+    local skillId = skill:getID()
+    local Para = GetMobByID(ID.mob.PARA)
+    local pos = target:getPos()
+
+    if
+        skillId ~= 308 and
+        skillId ~= 309 and
+        skillId ~= 314 and
+        skillId ~= 315 and
```

Comment:
These should be added and named in `enums/mob_skills.lua`

This skillId + HP condition is repeated in all of these blocks, you can extract it out into a local helper function to make this if/else tree more readable. You can make the whole thing more readable by switching on the value of `para:getLocalVar('spawnAdds')`, or doing table key lookups etc.

Reply:
Added to enums/mob_skills.lua and check was rewritten into a function.